### PR TITLE
Torsion angle loss

### LINKED ISSUE #66: Torsion angle loss

### PR DESCRIPTION
**Pull request by 0x00b1**
Closes #66

### CODE REVIEW

File: tests/beignet/nn/functional/test__torsion_angle_loss.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import beignet.nn.functional
+import torch
+
+
+def test_torsion_angle_loss():
```
Comment (0x00b1) [inline tests/beignet/nn/functional/test__torsion_angle_loss.py:5]: @kleinhenz Just a smoke test. I will revisit with improved unit tests after I implement the scaffolding for all the losses.

File: src/beignet/nn/functional/_torsion_angle_loss.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    Parameters
+    ----------
+    input
+    target
+
+    Returns
+    -------
+
+    """
+    a = input / torch.norm(input, dim=-1, keepdim=True)
```
Comment (0x00b1) [inline src/beignet/nn/functional/_torsion_angle_loss.py:19]: Should this be clamped? 🤔

File: src/beignet/nn/functional/_torsion_angle_loss.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+from typing import Tuple
+
+import torch
+from torch import Tensor
+
+
+def torsion_angle_loss(input, target: Tuple[Tensor, Tensor]) -> Tensor:
```
Comment (kleinhenz) [inline src/beignet/nn/functional/_torsion_angle_loss.py:7]: Can we call this something like `alphafold2_torsion_angle_loss` and add a reference to alphafold2 supplement algorithm 27 in the docstring?

Comment (0x00b1) [reply src/beignet/nn/functional/_torsion_angle_loss.py:7]: Of course. I will add a docstring!

As far as renaming, are there multiple torsion angle losses?

File: src/beignet/nn/functional/_torsion_angle_loss.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+from typing import Tuple
```
Comment (kleinhenz) [inline src/beignet/nn/functional/_torsion_angle_loss.py:1]: We could use [PEP 585 ](https://peps.python.org/pep-0585/) instead so the hint would be `tuple[Tensor, Tensor]`. I find it slightly cleaner but not a big deal either way.

File: src/beignet/nn/functional/_torsion_angle_loss.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    Returns
+    -------
+
+    """
+    a = input / torch.norm(input, dim=-1, keepdim=True)
+
+    b, c = target
+
+    x = torch.mean(
```
Comment (kleinhenz) [inline src/beignet/nn/functional/_torsion_angle_loss.py:23]: Can we call `x` and `y` something slightly more descriptive like `loss_torsion` and `loss_anglenorm` to make it easier to compare with the description in the supplement?

File: src/beignet/nn/functional/_torsion_angle_loss.py (hunk 4 above)
Comment (kleinhenz) [reply src/beignet/nn/functional/_torsion_angle_loss.py:1]: Looking into this more it looks like those hints are actually being deprecated and removed once python 3.9 is EOL
